library/ssl_tls.c

```diff
@@ -11271,22 +11267,40 @@ static int ssl_write_real( mbedtls_ssl_context *ssl,
          * copy the data into the internal buffers and setup the data structure
          * to keep track of partial writes
          */
-        ssl->out_msglen  = len;
+        ssl->out_msglen = len;
         ssl->out_msgtype = MBEDTLS_SSL_MSG_APPLICATION_DATA;
-        mbedtls_platform_memcpy( ssl->out_msg, buf, len );
+        mbedtls_platform_memcpy(ssl->out_msg, buf, len);
 
-        if( ( ret = mbedtls_ssl_write_record( ssl, SSL_FORCE_FLUSH ) ) != 0 )
+#if defined(MBEDTLS_FI_COUNTERMEASURES) && !defined(MBEDTLS_SSL_CBC_RECORD_SPLITTING)
```

Comment:
What is the connection between `MBEDTLS_FI_COUNTERMEASURES` and `MBEDTLS_SSL_CBC_RECORD_SPLITTING`? It is not mentioned anywhere.

Reply:
@AndrzejKurek , In general, there is no connection between MBEDTLS_FI_COUNTERMEASURES and MBEDTLS_SSL_CBC_RECORD_SPLITTING, but in this specific countermeasure we are saving pointers to user data, and in case of splinting, user data pointers will be changed after splitting operation. So this countermeasure cannot be supported with splitting.  I will add comment to explain it